hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1018,11 +1017,37 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Set SupportsMultiArchNodePools capable condition
+	{
+		condition := meta.FindStatusCondition(hcluster.Status.Conditions, hyperv1.SupportsMultiArchNodePools)
+
+		if condition == nil || condition.ObservedGeneration != hcluster.Generation || condition.Status != metav1.ConditionTrue {
+			condition := metav1.Condition{
```

Comment:
Why is this a condition vs just doing all validating in flight? This is actually using bool state but then still need in flight validation to match the nodepool and HC arch when not multi arch.
If we choose persisting state why not use a status field with enum for cpus so there's no need for inflight operations at all?